src/apis/web3api.py

```diff
@@ -0,0 +1,215 @@
+"""
+Web3API for fetching relevant data using the web3 library.
+"""
+from os import getenv
+from typing import Dict, List, Tuple, Any, Optional
```

Comment:
Similar to above, try to eliminate all but Any and Optional here. For Dict, List and Tuple, these can all be replaced with their lower case "primitives". In an idea world, you would also be able to eliminate the use of Any from the project as well, but I won't force you at this point - as long as you capture this as an issue.

Reply:
I opened issue #51.